[BUG] Failing to connect to an OpenSearch cluster with a bad username/password gives unhelpful error

**Describe the bug**

Running Data Prepper with a bad username/password combination yields the following error.

```
2023-05-08T19:29:16,393 [main] ERROR org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink - Failed to initialize OpenSearch sink.
2023-05-08T19:29:16,393 [Thread-2] ERROR org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink - Failed to initialize OpenSearch sink.
Exception in thread "main" jakarta.json.stream.JsonParsingException: Jackson exception: Unrecognized token 'Unauthorized': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (ByteArrayInputStream); line: 1, column: 13]
	at org.opensearch.client.json.jackson.JacksonJsonpParser.convertException(JacksonJsonpParser.java:97)
	at org.opensearch.client.json.jackson.JacksonJsonpParser.fetchNextToken(JacksonJsonpParser.java:104)
	at org.opensearch.client.json.jackson.JacksonJsonpParser.next(JacksonJsonpParser.java:131)
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:82)
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:92)
	at org.opensearch.client.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:56)
	at org.opensearch.client.transport.rest_client.RestClientTransport.getHighLevelResponse(RestClientTransport.java:271)
	at org.opensearch.client.transport.rest_client.RestClientTransport.performRequest(RestClientTransport.java:143)
	at org.opensearch.client.opensearch.cluster.OpenSearchClusterClient.getSettings(OpenSearchClusterClient.java:282)
	at org.opensearch.dataprepper.plugins.sink.opensearch.index.AbstractIndexManager.checkISMEnabled(AbstractIndexManager.java:187)
	at org.opensearch.dataprepper.plugins.sink.opensearch.index.AbstractIndexManager.checkAndCreateIndexTemplate(AbstractIndexManager.java:208)
	at org.opensearch.dataprepper.plugins.sink.opensearch.index.AbstractIndexManager.setupIndex(AbstractIndexManager.java:203)
	at org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink.doInitializeInternal(OpenSearchSink.java:174)
	at org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSink.doInitialize(OpenSearchSink.java:139)
	at org.opensearch.dataprepper.model.sink.AbstractSink.initialize(AbstractSink.java:39)
	at org.opensearch.dataprepper.pipeline.Pipeline.isReady(Pipeline.java:194)
	at org.opensearch.dataprepper.DataPrepper.execute(DataPrepper.java:93)
	at org.opensearch.dataprepper.DataPrepperExecute.main(DataPrepperExecute.java:42)
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Unauthorized': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (ByteArrayInputStream); line: 1, column: 13]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2418)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:759)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3693)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleUnexpectedValue(UTF8StreamJsonParser.java:2781)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:907)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:793)
	at org.opensearch.client.json.jackson.JacksonJsonpParser.fetchNextToken(JacksonJsonpParser.java:102)
	... 16 more
```

**To Reproduce**
Steps to reproduce the behavior:
1. Change the `examples/trace_analytics_no_ssl_2x.yml` file by giving `badpassword` in the `password` fields.
2. Run the Jaeger Hotrod example
3. Check Data Prepper logs (`docker logs -f data-prepper`)
4. See error

**Expected behavior**

Ideally, the error from OpenSearch should be visible.

Additionally, I'd like to see the username I provided. For example: "Unable to authenticate with username `admin`. Check your username and password."

**Additional context**

This is similar to #2655, but for open-source OpenSearch domains.

--- a/settings.gradle
+++ b/settings.gradle
@@ -26,7 +26,7 @@ dependencyResolutionManagement {
             library('protobuf-util', 'com.google.protobuf', 'protobuf-java-util').versionRef('protobuf')
             version('opentelemetry', '0.16.0-alpha')
             library('opentelemetry-proto', 'io.opentelemetry.proto', 'opentelemetry-proto').versionRef('opentelemetry')
-            version('opensearchJava', '2.2.0')
+            version('opensearchJava', '2.5.0')
             library('opensearch-java', 'org.opensearch.client', 'opensearch-java').versionRef('opensearchJava')
             version('opensearch', '1.3.8')
             library('opensearch-client', 'org.opensearch.client', 'opensearch-rest-client').versionRef('opensearch')

